### PR TITLE
# Replication start and replication streams lambdas

### PR DESCRIPTION
**Pull request by CCristi**

### CODE REVIEW

**Comment by devs-deep** [comment]
:white_check_mark: coverage increased from 0% to 100%

File: src/deep-blue-green/backend/src/replication/start/bootstrap.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+const AWS = require('aws-sdk');
+const CHUNK_SIZE = 100;
+
+exports.handler = (event, context, callback) => {
+  const tableManager = new TableManager(new AWS.DynamoDB(), event.Table);
+  const backFillExecutor = new BackFillExecutor(tableManager, event.Hash);
+  const scanPayload = {};
+
+  scanPayload[':hash'] = {
+    S: event.Hash,
```
Comment (AlexanderC) [inline src/deep-blue-green/backend/src/replication/start/bootstrap.js]: Please use the following node module for all build/read ops in/from DDB: https://github.com/CascadeEnergy/dynamoDb-marshaler

Comment (AlexanderC) [inline src/deep-blue-green/backend/src/replication/start/bootstrap.js]: And no, it will not slow down anything!

Comment (AlexanderC) [inline src/deep-blue-green/backend/src/replication/start/bootstrap.js]: For sure!!!

File: src/deep-blue-green/backend/src/replication/start/bootstrap.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+      .then(itemsProcessed => {
+        if (itemsProcessed > 0) {
+          console.log(`"${itemsProcessed}" items have been marked`);
+          backFillProcessor();
+        } else {
+          callback(null, {});
+        }
+      })
+      .catch(e => {
+        callback({
```
Comment (AlexanderC) [inline src/deep-blue-green/backend/src/replication/start/bootstrap.js]: Please follow the following standard for callbacks: `callback(error || null, null || result)` where `error instanceof Error`

File: src/deep-blue-green/backend/src/replication/start/bootstrap.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+          error: e,
+          stack: e.stack.toString(),
+        }, null);
+      });
+  };
+
+  backFillProcessor();
+};
+
+class BackFillExecutor {
```
Comment (AlexanderC) [inline src/deep-blue-green/backend/src/replication/start/bootstrap.js]: Could you put each class in a separate file?

File: src/deep-blue-green/backend/src/replication/start/bootstrap.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+
+    payload.TableName = this._tableName;
+    payload.FilterExpression = filterExpression;
+    payload.ProjectionExpression = fields.map(f => `#${f}`).join(',');
+    payload.ExpressionAttributeNames = fields.reduce((obj, f) => ((obj[`#${f}`] = f) && obj), {});
+    payload.ExpressionAttributeValues = filterValues;
+    payload.ReturnConsumedCapacity = 'NONE';
+
+    return payload;
+
```
Comment (AlexanderC) [inline src/deep-blue-green/backend/src/replication/start/bootstrap.js]: Please remove newline here

File: src/deep-blue-green/backend/src/replication/start/bootstrap.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+  get Id() {
+    return this._data.Id;
+  }
+
+  get data() {
+    return this._data;
+  }
+}
+
+class Serializer {
```
Comment (AlexanderC) [inline src/deep-blue-green/backend/src/replication/start/bootstrap.js]: The best marshaler i've ever seen!

**Pull request merged**